autogen/agentchat/contrib/quantifier_agent.py

```diff
@@ -0,0 +1,52 @@
+from typing import Callable, Dict, Literal, Optional, Union
+
+from autogen.agentchat.conversable_agent import ConversableAgent
+from autogen.runtime_logging import logging_enabled, log_new_agent
+
+
+class QuantifierAgent(ConversableAgent):
+    """
+    An agent for quantifing the performance of a system using the provided criteria.
+    """
+
+    DEFAULT_SYSTEM_MESSAGE = """"You are a helpful assistant. You quantify the output of different tasks based on the given criteria.
+    The criterion is given in a dictionary format where each key is a dintinct criteria.
+    The value of each key is a dictionary as follows {"description": criteria description , "accepted_values": possible accepted inputs for this key}
+    You are going to quantify each of the crieria for a given task based on the task description.
+    Return a dictionary where the keys are the criteria and the values are the assessed performance based on accepted values for each criteria.
+    Return only the dictionary."""
+
+    DEFAULT_DESCRIPTION = "An AI agent for quantifing the performance of a system using the provided criteria."
+
+    def __init__(
+        self,
+        name="quantifier",
+        system_message: Optional[str] = DEFAULT_SYSTEM_MESSAGE,
+        llm_config: Optional[Union[Dict, bool]] = None,
+        max_consecutive_auto_reply: Optional[int] = None,
+        human_input_mode: Optional[str] = "NEVER",
+        description: Optional[str] = DEFAULT_DESCRIPTION,
+        **kwargs,
+    ):
+        """
+        Args:
+            - name (str): agent name.
+            - system_message (str): system message for the ChatCompletion inference.
+                Please override this attribute if you want to reprogram the agent.
+            - llm_config (dict or False or None): llm inference configuration.
```

Comment:
quantifier seems to work better with json mode but that is controlled by the llm_config. Maybe its worth mentioning that with a docstring comment.

Reply:
I added a little bit to be consistent with other agent's documentation.

Reply:
a link to the openai docs is too generic - here a user would benefit specifically by turning on json mode. I agree consistency is good but if classes have specialized behavior then they can't be consistent with each other. This also might be some autogen-ism that we're hitting bc json mode hasn't been formalized as a first class citizen afaik.